helix-gateway/src/main/java/org/apache/helix/gateway/channel/GatewayServiceChannelConfig.java

```diff
@@ -0,0 +1,173 @@
+package org.apache.helix.gateway.channel;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+import static org.apache.helix.gateway.api.constant.GatewayServiceDefaultConfig.*;
+
+public class GatewayServiceChannelConfig {
+  public enum ChannelType {
+    GRPC_SERVER, POLL_GRPC, SHARED_FILE
```

Comment:
use the new formatter of Helix and make it one per line.

Reply:
Also let's define right axis of the these types:

grpc as long lived (push),
grpc as short lived (poll),
file based (poll).

Reply:
Long lived grpc stream may not necessary be push mode. We could maintain a stream connection with some endpoint to poll heath state and shard state. 
Similarly, push mode can be unary grpc as well.

Reply:
At least, let's make name consistent:

GRPC_PUSH
GRPC_POLL
FILE_POLL